sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnInvokers.java

```diff
@@ -501,8 +493,8 @@ protected StackManipulation beforeDelegation(MethodDescription instrumentedMetho
 
     private final MethodDescription createUserCodeException;
 
-    public UserCodeMethodInvocation(
-        Integer returnVarIndex,
+    UserCodeMethodInvocation(
+        @Nullable Integer returnVarIndex,
```

Comment:
This may be be better written using `Optional<Integer>` since it makes the contract more obvious.